Assets/Scripts/VFX/RenderSettings/RenderSettingsManager.cs

```diff
@@ -0,0 +1,193 @@
+		timeOfDay = Mathf.Clamp01( timeOfDay );
+		
+		daylightIntensity = Mathf.Cos( timeOfDay * 2.0f * Mathf.PI ) * -0.5f + 0.5f;
+		
+		TimeLightingSettings.Lerp( _currentLightingSettings.nightSettings,
+		                           _currentLightingSettings.daySettings,
+		                           daylightIntensity,
+		                           ref _currentTimeRenderSettings );
+	}
+
+	void UpdateFogGradientTexture()
+	{
+		Color[] gradientColors = new Color[128];
+		for( int i = 0; i < 128; i++ )
+		{
+			gradientColors[i] = _currentTimeRenderSettings.fogGradient.Evaluate( i/128f );
+		}
+		gradientTexture.SetPixels( gradientColors );
+		gradientTexture.Apply();
+	}
+
+	void UpdateRenderSettings()
+	{
+		UpdateTimeOfDay();
+
+		if( globalFog.fogMaterial != null )
+		{
+			UpdateFogGradientTexture();
+			globalFog.fogMaterial.SetTexture( "_FogGradientTex", gradientTexture );
+		}
+		
+		RenderSettings.skybox.SetColor( _skyboxTintPropertyID, _currentTimeRenderSettings.skyColor );
+		RenderSettings.fogDensity = _currentTimeRenderSettings.fogDensity;
+		
+		_directionalLight.color = _currentTimeRenderSettings.lightColor;
+		_directionalLight.intensity = _currentTimeRenderSettings.lightIntensity;
+	}
+
+	public static void TransitionRenderSettings( LightingSettings newRenderSettings, float shiftTime )
+	{
+		instance.StopAllCoroutines(); // TODO: Make this not a sledgehammer solution
+		instance.StartCoroutine( instance.TransitionRenderSettingsRoutine( newRenderSettings, shiftTime ) );
+	}
+
+	IEnumerator TransitionRenderSettingsRoutine( LightingSettings newRenderSettings, float settingShiftTime )
+	{
+		LightingSettings oldRenderSettings = _currentLightingSettings;
+
+		float settingShiftTimer = 0f;
+		while ( settingShiftTimer < settingShiftTime )
+		{
+			TimeLightingSettings.Lerp( oldRenderSettings.daySettings, 
+			                           newRenderSettings.daySettings,
+			                           settingShiftTimer/settingShiftTime,
+			                           ref _currentLightingSettings.daySettings );
+
+			TimeLightingSettings.Lerp( oldRenderSettings.nightSettings,
+			                           newRenderSettings.nightSettings,
+			                           settingShiftTimer/settingShiftTime,
+			                           ref _currentLightingSettings.nightSettings );
+
+			UpdateRenderSettings();
+
+			settingShiftTimer += Time.deltaTime;
+			yield return 0;
+		}
+
+		_currentLightingSettings = newRenderSettings;
+	}
+
+	public static void SetRenderSettings( LightingSettings lightSettings )
+	{
+		instance._currentLightingSettings = lightSettings;
+		instance.UpdateRenderSettings();
+	}
+
+	public static void SetToNearestZone( Vector3 position )
+	{
+		SetRenderSettings( GetNearestRenderSettings( position ) );
+	}
+
+	public static void TransitionToNearestZone( Vector3 position, float time )
+	{
+		TransitionRenderSettings( GetNearestRenderSettings( position ), time );
+	}
```

Comment:
I use this in the cutscene to hard set the render settings at the start of each shot